packages/blade/src/components/Tour/Tour.web.tsx

```diff
@@ -0,0 +1,199 @@
+import { useTheme } from '~utils';
+import { useIsomorphicLayoutEffect } from '~utils/useIsomorphicLayoutEffect';
+
+const Tour = ({
+  steps,
+  activeStep,
+  isOpen,
+  onFinish,
+  onOpenChange,
+  onStepChange,
+  children,
+}: TourProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const [refIdMap, setRefIdMap] = useState(new Map<string, React.RefObject<HTMLElement>>());
+  const [size, setSize] = useState<TourMaskRect>({
+    x: 0,
+    y: 0,
+    height: 0,
+    width: 0,
+  });
+
+  // delayed state is used to let the transition finish before reacting to the state changes
+  const delayedActiveStep = useDelayedState(activeStep, transitionDelay);
+  const delayedSize = useDelayedState(size, transitionDelay);
+  // keep track of when we are transitioning between steps
+  const isTransitioning = useIsTransitioningBetweenSteps(activeStep, transitionDelay);
+
+  const currentStepRef = refIdMap.get(steps[delayedActiveStep]?.name);
+  const intersection = useIntersectionObserver(currentStepRef!, {
+    threshold: 0.5,
+  });
+
+  // main step logic
+  const totalSteps = steps.length;
+  const currentStepData = useMemo(() => {
+    return steps[activeStep];
+  }, [activeStep, steps]);
+
+  const goToNext = useCallback(() => {
+    let nextState = activeStep + 1;
+    if (nextState >= steps.length) {
+      nextState = steps.length - 1;
+    }
+    onStepChange?.(nextState);
+  }, [activeStep, onStepChange, steps.length]);
+
+  const goToPrevious = useCallback(() => {
+    let nextState = activeStep - 1;
+    if (nextState < 0) {
+      nextState = 0;
+    }
+    onStepChange?.(nextState);
+  }, [activeStep, onStepChange]);
+
+  const stopTour = useCallback(() => {
+    onFinish?.();
+  }, [onFinish]);
+
+  const attachStep = useCallback((id: string, ref: React.RefObject<HTMLElement>) => {
+    if (!ref) return;
+    setRefIdMap((prev) => {
+      return new Map(prev).set(id, ref);
+    });
+  }, []);
+
+  const removeStep = useCallback((id: string) => {
+    setRefIdMap((prev) => {
+      const newMap = new Map(prev);
+      newMap.delete(id);
+      return newMap;
+    });
+  }, []);
+
+  const updateMaskSize = useCallback(() => {
+    const ref = refIdMap.get(steps[activeStep]?.name);
+    if (!ref?.current) return;
+
+    const rect = ref.current.getBoundingClientRect();
+    setSize({
+      x: rect.x,
+      y: rect.y,
+      width: rect.width,
+      height: rect.height,
+    });
+  }, [activeStep, refIdMap, steps]);
+
+  const scrollToStep = useCallback(() => {
+    const ref = refIdMap.get(steps[delayedActiveStep]?.name);
+    if (!ref?.current) return;
+
+    // If the element is already in view, don't scroll
+    if (intersection?.isIntersecting) return;
+
+    smoothScroll(ref.current, {
+      behavior: 'smooth',
+      block: 'center',
+      inline: 'center',
+    })
+      .then(() => {
+        updateMaskSize();
+      })
+      .finally(() => {
+        // do nothing
+      });
```

Comment:
remove if not needed

Reply:
Eslint is for floating [promises](https://typescript-eslint.io/rules/no-floating-promises/). Might as well just add this.